sqswatcher/sqswatcher.py

```diff
@@ -122,87 +118,106 @@ def pollQueue(scheduler, q, t, proxy_config):
+                        event_type = message_attrs.get('detail-type')
                     except KeyError:
-                        log.warn("Unable to read message. Deleting.")
+                        log.warning("Unable to read message. Deleting.")
                         message.delete()
                         break
 
-                log.info("eventType=%s" % eventType)
-                if eventType == 'autoscaling:TEST_NOTIFICATION':
+                log.info("eventType=%s" % event_type)
+                if event_type == 'autoscaling:TEST_NOTIFICATION':
                     message.delete()
-                elif eventType == 'parallelcluster:COMPUTE_READY':
-                    instanceId = message_attrs.get('EC2InstanceId')
+                elif event_type == 'parallelcluster:COMPUTE_READY':
+                    instance_id = message_attrs.get('EC2InstanceId')
                     slots = message_attrs.get('Slots')
-                    log.info("instanceId=%s" % instanceId)
+                    log.info("instanceId=%s" % instance_id)
                     ec2 = boto3.resource('ec2', region_name=region, config=proxy_config)
 
                     retry = 0
                     wait = 15
                     while retry < 3:
                         try:
-                            instances = ec2.instances.filter(InstanceIds=[instanceId])
+                            instances = ec2.instances.filter(InstanceIds=[instance_id])
                             instance = next(iter(instances or []), None)
 
                             if not instance:
-                                log.warning("Unable to find running instance %s." % instanceId)
+                                log.error("Unable to find running instance %s." % instance_id)
                             else:
                                 hostname = instance.private_dns_name.split('.')[:1][0]
-                                log.info("Adding Hostname: %s" % hostname)
-                                s.addHost(hostname=hostname, cluster_user=cluster_user, slots=slots)
-
-                                t.put_item(Item={
-                                    'instanceId': instanceId,
-                                    'hostname': hostname
-                                })
-
-                            message.delete()
+                                if hostname:
+                                    log.info("Adding hostname: %s" % hostname)
+                                    scheduler_module.addHost(hostname=hostname, cluster_user=cluster_user, slots=slots)
+
+                                    table.put_item(Item={
+                                        'instanceId': instance_id,
+                                        'hostname': hostname
+                                    })
+                                else:
+                                    log.error("Unable to get the hostname for the instance %s." % instance_id)
```

Comment:
So here when it can't get the hostname, it deletes the sqs message and continues?